src/device.rs

```diff
@@ -41,17 +41,27 @@ impl Device {
                             if !peripheral.is_connected().await? {
                                 peripheral.connect().await?;
                             }
-                            let _ = peripheral.discover_characteristics().await?;
 
-                            let supports_compression = match name {
-                                "MX10" => false,
-                                _ => true,
-                            };
+                            println!("Connected to {}", name);
+                            println!("Discovering services...");
+                            peripheral.discover_services().await?;
+
+                            let supports_compression = name != "MX10";
```

Comment:
```suggestion
                            let supports_compression = match name {
                                "MX10" => false,
                                _ => true,
                            };
```
While more verbose, I intentionally used a match here to be able to easily add more devices in the future.